Userland/sort.cpp

```diff
@@ -45,17 +46,15 @@ int main(int argc, char** argv)
     Vector<String> lines;
 
     for (;;) {
-        char* buffer = nullptr;
-        ssize_t buflen = 0;
-        errno = 0;
-        buflen = getline(&buffer, nullptr, stdin);
-        if (buflen == -1 && errno != 0) {
+        auto result = Core::IODevice::get_line(stdin, false);
+        if (result.is_error()) {
+            if (result.error() == Core::IODevice::Eof)
+                break;
+            errno = result.error();
             perror("getline");
```

Comment:
Not a fan of assigning to `errno` like this just so we can use `perror`.
How about this instead:

```cpp
warnln("getline: {}", strerror(result.error());
```

Reply:
Eh, it did feel somewhat silly when I did that, I guess just using `strerror()` is better...until someone changes `perror` to do something different?

Reply:
`perror` is a standard C library function though, so it shouldn't be changed to do something different :)

Reply:
Fair enough.